flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/internal/TableEnvironmentImpl.java

```diff
@@ -461,6 +463,23 @@ public boolean dropTemporaryFunction(String path) {
         return functionCatalog.dropTemporaryCatalogFunction(unresolvedIdentifier, true);
     }
 
+    @Override
+    public void addJar(String jarPath) {
+        try {
+            resourceManager.registerJarResources(
```

Comment:
Do we need save jar into catalog?
Or added Jar always work in session scope?

Reply:
It is not related to catalog.